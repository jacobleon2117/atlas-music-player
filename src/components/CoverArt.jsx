import React from "react";
import placeholder from "../assets/placeholder.svg";

const CoverArt = ({ imageUrl }) => {
  return (
    <div className="h-48 w-48 overflow-hidden rounded-lg bg-gray-100 shadow-lg">
      <img
        src={imageUrl || placeholder}
        alt="Album Cover"
        className="h-full w-full object-cover"
      />
    </div>
  );
};

export default CoverArt;
