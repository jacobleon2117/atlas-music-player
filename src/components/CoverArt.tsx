import { useState } from 'react';

interface CoverArtProps {
  songId: string;
  coverUrl: string | null;
  title: string;
}

export default function CoverArt({ songId, coverUrl, title }: CoverArtProps) {
  const [lyrics, setLyrics] = useState<string>('');
  const [isLoading, setIsLoading] = useState(false);
  const [showLyrics, setShowLyrics] = useState(false);

  const fetchLyrics = async () => {
    if (!lyrics && !isLoading) {
      setIsLoading(true);
      try {
        const response = await fetch(`/api/v1/lyrics/${songId}`);
        if (!response.ok) throw new Error('Failed to fetch lyrics');
        const data = await response.json();
        setLyrics(data.lyrics);
      } catch (error) {
        console.error('Error fetching lyrics:', error);
      } finally {
        setIsLoading(false);
      }
    }
  };

  return (
    <div 
      className="relative h-[400px] w-[400px]"
      onMouseEnter={() => {
        setShowLyrics(true);
        fetchLyrics();
      }}
      onMouseLeave={() => setShowLyrics(false)}
    >
      {coverUrl ? (
        <img
          src={coverUrl}
          alt={`Album artwork for ${title}`}
          className="h-full w-full rounded-md object-cover"
        />
      ) : (
        <div className="h-full w-full bg-gray-200 rounded-md flex items-center justify-center">
          <span className="text-gray-500">No Cover</span>
        </div>
      )}

      {showLyrics && (
        <div className="absolute inset-0 overflow-auto bg-black/75 p-4 text-white opacity-0 transition-opacity hover:opacity-100">
          {isLoading ? (
            <div className="flex h-full items-center justify-center">
              <div className="h-6 w-6 animate-spin rounded-full border-2 border-white border-t-transparent" />
            </div>
          ) : (
            <pre className="whitespace-pre-wrap font-inter text-sm">
              {lyrics}
            </pre>
          )}
        </div>
      )}
    </div>
  );
}
