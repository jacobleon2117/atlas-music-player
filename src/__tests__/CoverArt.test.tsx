import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import CoverArt from '../components/CoverArt';

describe('CoverArt', () => {
  it('renders with all props', () => {
    const { container } = render(
      <CoverArt
        songId="test-id"
        coverUrl="test-cover.jpg"
        title="Test Song"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with minimal props', () => {
    const { container } = render(
      <CoverArt
        songId="test-id"
        coverUrl=""
        title=""
      />
    );
    expect(container).toMatchSnapshot();
  });
});