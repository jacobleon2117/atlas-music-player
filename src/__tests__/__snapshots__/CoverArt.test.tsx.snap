// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt > renders with all props 1`] = `
<div>
  <div
    class="relative h-[400px] w-[400px]"
  >
    <img
      alt="Album artwork for Test Song"
      class="h-full w-full rounded-md object-cover"
      src="test-cover.jpg"
    />
  </div>
</div>
`;

exports[`CoverArt > renders with minimal props 1`] = `
<div>
  <div
    class="relative h-[400px] w-[400px]"
  >
    <div
      class="h-full w-full bg-gray-200 rounded-md flex items-center justify-center"
    >
      <span
        class="text-gray-500"
      >
        No Cover
      </span>
    </div>
  </div>
</div>
`;
